main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -33,13 +33,41 @@ object CompleteProvider {
     * Returns a list of auto-complete suggestions.
     */
   def autoComplete(uri: String, pos: Position, line: Option[String], word: Option[String])(implicit index: Index, root: TypedAst.Root): Iterable[CompletionItem] = {
-    // Ordered by priority.
-    getVarSuggestions(uri, pos, line, word) ++
-      getDefAndSigSuggestions(uri, pos, line, word) ++
-      getInstanceSuggestions(uri, pos, line, word) ++
-      getWithSuggestions(uri, pos, line, word) ++
-      getKeywordCompletionItems(line, word) ++
-      getSnippetCompletionItems(line, word)
+    word match {
+      case None => Nil;
+      case Some(w) => 
+        if (w.startsWith("@")) {
+          getAnnotationCompletionItems(line, word)
+        } else {
+          // Ordered by priority.
+          getVarSuggestions(uri, pos, line, word) ++
+            getDefAndSigSuggestions(uri, pos, line, word) ++
+            getInstanceSuggestions(uri, pos, line, word) ++
+            getWithSuggestions(uri, pos, line, word) ++
+            getKeywordCompletionItems(line, word) ++
+            getSnippetCompletionItems(line, word)
+        }
+    }
+  }
+
+  /**
+    * Returns a list of annotation completion items.
+    */
+  private def getAnnotationCompletionItems(line: Option[String], word: Option[String]): List[CompletionItem] = {
+    List(
+      CompletionItem("benchmark", "benchmark ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("test", "test ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Deprecated", "Deprecated ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Experimental", "Experimental ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Internal", "Internal ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("ParallelWhenPure", "ParallelWhenPure ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Parallel", "Parallel ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("LazyWhenPure", "LazyWhenPure ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Lazy", "Lazy ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Space", "Space ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Time", "Time ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil),
+      CompletionItem("Unsafe", "Unsafe ", None, Some("annotation"), CompletionItemKind.Keyword, InsertTextFormat.PlainText, Nil)
+    )
```

Comment:
The above approach is reasonable, but I would remove the following:

- Internal
- Unsafe

since we are not sure where we are going with those.

Reply:
Oh, and I tihnk all of the completion items should start with `@`

Reply:
Regarding starting with `@`, I agree that that's probably how it should be, but as I said in Gitter, if I do that then when the completion is inserted by VSCode, the @ ends up being doubled.

My guess is that something within SemanticTokensProvider probably needs updating (I think that that's where VSCode is getting its tokenisation from?). But I'm not sure where to start here?

Reply:
I am not sure what the solution is, but the SemanticTokensProvider is completely separate from auto complete.

Reply:
(i.e. they have no interaction whatsoever in VSCode)

Reply:
So this code simply returns a list of `CompletionItem`s (it ignores the parameters). So presumably VSCode is doing some kind of tokenisation and matching to decide which of the `CompletionItem`s are applicable?

Right now, it doesn't seem to be considering an `@` character to be part of the token under consideration? Presumably something is telling VSCode what constitutes a token for Flix? I can't find anything within https://github.com/flix/vscode-flix which defines this, so either it's something within VSCode itself, or something within the language server?

Reply:
We have often found that VSCode makes all kinds of "interpretations". You may think its just an API, but often it will "re-interpret" the data you are sending back. Its possible that it treats `@` differently in some kind of implementation specific way. (its also possible we have a mistake). But this would not be first time that VSCode does something slightly different than from what you would think the spec says.

Reply:
(Auto-completion does not use any kind of tokenization really).

Reply:
What happens if you change "Some(annotation)" to "Some(keyword)" does it behave differently?

Reply:
Yeah, I tried changing `Some(annotation)`. It makes no functional difference (it's just help text that's displayed if you click on the `>` in the UI).

Reply:
Ok